src/crimson/os/seastore/omap_manager/btree/omap_btree_node_impl.h

```diff
@@ -28,20 +30,91 @@ namespace crimson::os::seastore::omap_manager {
 
 struct OMapInnerNode
   : OMapNode,
-    StringKVInnerNodeLayout {
+    StringKVInnerNodeLayout,
+    ParentNode<OMapInnerNode, std::string>,
+    ChildNode<OMapInnerNode, OMapInnerNode, std::string> {
   using OMapInnerNodeRef = TCachedExtentRef<OMapInnerNode>;
-  using internal_iterator_t = const_iterator;
+  using internal_const_iterator_t = const_iterator;
+  using internal_iterator_t = iterator;
+  using parent_node_t = ParentNode<OMapInnerNode, std::string>;
+  using base_child_t = BaseChildNode<OMapInnerNode, std::string>;
+  using child_node_t = ChildNode<OMapInnerNode, OMapInnerNode, std::string>;
+  static constexpr uint32_t CHILD_VEC_UNIT = 128;
 
   explicit OMapInnerNode(ceph::bufferptr &&ptr)
-    : OMapNode(std::move(ptr)) {
-    this->set_layout_buf(this->get_bptr().c_str());
-  }
+    : OMapNode(std::move(ptr)),
+      StringKVInnerNodeLayout(get_bptr().c_str()),
+      parent_node_t(p2roundup(get_size(), CHILD_VEC_UNIT)) {}
```

Comment:
Let's follow the exact logic to call `parent_node_t(0)` and call `sync_children_capacity()` in the cstor body.

Otherwise we will need to remember these places when want to adjust `sync_children_capacity()` in the future.